### DISCUSSION TITLE
Settings order in IDEA UnitTestBot

### DISCUSSION THREAD

**Discussion by alisevych** [UnitTestBot/UTBotJava#2177] [Polls]
Please vote for the better ordering.

How it looks now (ordered by element types):

![image](https://user-images.githubusercontent.com/37301492/233946105-b1a333ac-d053-4392-945c-ed91c5881b9b.png)

Second option (ordered by sense):

![image](https://user-images.githubusercontent.com/37301492/233946449-8ccff714-766b-47e6-a2fa-d6a9504329c0.png)

Details:
- #1857

**Comment by tyuldashev** [comment]
I like conceptual grouping option better, but would like to discuss UI and layout in case of changes.

**Reply by alisevych** [reply]
Ok, can you please prepare suggestions what you'd like to improve?

**Reply by tyuldashev** [reply]
I thought about two things:
 * Move "Experimental languages" checkbox on the same line as "Generated test language" (at the right)
 * Move "Display detected errors" to after "Experimental language support". So it is not inside code generation settings.

**Comment by olganaumenko** [comment]
I think we should follow the IntelliJ Platform UI Guidelines: https://jetbrains.github.io/ui/principles/layout/

I would use vertical insets to break the controls into groups (without group headers):

Generated test language:
Experimental languages support

Tests with exceptions
Oveflow detection
Display detected errors in the Problems tool window

Enable summaries generation
Javadoc comment style

Force mocking static methods
Classes to be forcedly mocked

Hanging test timeout:
Test generation method

**Comment by alisevych** [comment]
Please check the text in **Suggested behavior** section in the task: 
- #1857
